views/js/runner/plugins/navigation/next/nextItemWarning.js

```diff
@@ -0,0 +1,155 @@
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2019 (original work) Open Assessment Technologies SA ;
+ */
+
+/**
+ * Test Runner Navigation Plugin : nextItemWarning
+ *
+ * @author Martin Nicholson <martin@taotesting.com>
+ */
+define([
+    'jquery',
+    'lodash',
+    'i18n',
+    'taoTests/runner/plugin',
+    'taoQtiTest/runner/helpers/currentItem',
+    'taoQtiTest/runner/plugins/navigation/next/dialogConfirmNext'
+], function ($, _, __, pluginFactory, currentItemHelper, dialogConfirmNext){
+    'use strict';
+
+    /**
+     * Returns the configured plugin
+     */
+    return pluginFactory({
+        name : 'nextItemWarning',
+
+        /**
+         * Initialize the plugin (called during runner's init)
+         */
+        init : function init(){
+            var self = this;
+            var testRunner = this.getTestRunner();
+            var testData = testRunner.getTestData();
+            var testConfig = testData.config || {};
+            var testStore = testRunner.getTestStore(); // we'll store user's checkbox choice in here
+            testStore.setVolatile(self.getName());
+
+            /**
+             * Provides different variants of message text
+             * @returns {String}
+             */
+            function getCustomNextMessage(action) {
+                var customNextMessage;
+                var itemPartiallyAnswered = currentItemHelper.isAnswered(testRunner, true);
+                if (! itemPartiallyAnswered) {
+                    customNextMessage = __('Are you sure you want to go to the next item? You will not be able to go back and provide an answer.');
+                }
+                else if (action === 'next') {
+                    customNextMessage = __('Are you sure you want to go to the next item? You will not be able to go back and change your answer.');
+                }
+                else if (action === 'skip') {
+                    customNextMessage = __('Are you sure you want to clear your answer and go to the next item? You will not be able to go back and provide an answer.');
+                }
+                return customNextMessage;
```

Comment:
shouldn't we have a default value ? or if `action` is required, please throw a `TypeError` if the function is called with incorrect arguments